### PR TITLE
Get properly cased project name

### PR DESCRIPTION
**Pull request by hugodahl**
Gets the project name properly cased based on the directory and file
structure information.

Introduce "using" statements to the forms, so that they are properly
disposed when closed.

### CODE REVIEW

**Comment by gandarez** [comment]
Hi @hugodahl 
Why do you need to validate if file/directory exists since a solution file always will be on disk. Show me a case of use.

File: WakaTimePackage.cs [hunk 1]
```diff
@@ -228,20 +228,43 @@ private static void MenuItemCallback(object sender, EventArgs e)
...
+
+            if (di.Parent != null)
+            {
+                return Path.Combine(
+                    GetExactPathName(di.Parent.FullName),
+                    di.Parent.GetFileSystemInfos(di.Name)[0].Name);
+            }
+            else
+            {
+                return di.Name.ToUpper();
```
Comment (alanhamlett) [inline WakaTimePackage.cs]: Why uppercase the project name?

**Comment by hugodahl** [comment]
This code is pretty much a direct copy of a utility method I have, so it's fairly defense, just-in-case type of situation.

The true intent is to consistently send the correct file casing project name to the server, in order to prevent false duplicates. Such as `My.Awesome.Project` and `my.awesome.project` which I have had happen.

This way, the name of the solution is always the same, based on the actual file case on disk.

File: WakaTimePackage.cs (hunk 1 above)
Comment (hugodahl) [inline WakaTimePackage.cs]: It doesn't uppercase the project name, the uppercase (`ToUpper()` call) is a fallback, usually used for the drive letter.

**Comment by gandarez** [comment]
@hugodahl I hevn't seen it happening since I started using the plugin. All my project data are logged correctly. I've been working in several projects both at home and at work.

**Comment by hugodahl** [comment]
Here's an example screenshot from the weekly email report, which shows a few such "duplicates". And these are all from the same machine, same user login, same path, etc.

screenshot 2015-06-11 21 44 03

**Comment by gandarez** [comment]
I'd suggest you to enable debugging into .wakatime.cfg file and then inspect all those lines to see if the problem is related to the plugin. Then we can point to which process is failing.

**Comment by PlatinumEagle** [comment]
Just FYI, I also suffer from the same duplicate project name that @hugodahl posted above. 

Not commenting on the proposed solution approach.

**Comment by alanhamlett** [comment]
@PlatinumEagle and @hugodahl, I've seen duplicate project names from two scenarios:

1. when working with the same project in multiple IDEs or text editors
2. when working with the same project on multiple machines

Because this problem isn't limited to Visual Studio, let's solve it on the server by suggesting merges of projects if they share the same name but different case.

I'll post back here when it's ready to use on the website.

**Comment by hugodahl** [comment]
Thanks for the feedback @alanhamlett!

**Comment by PlatinumEagle** [comment]
Hi Alan,
Just FYI, I get this even when only working on the same system.
As you are envisioning the feature, would the "merge" be persistent (so that you don't have to keep going back and merging additional data)?

**Comment by alanhamlett** [comment]
@PlatinumEagle yes, it would be persistent into the future.

The way it will be implemented, it will also allow renaming projects as a side benefit.

**Comment by PlatinumEagle** [comment]
Great. Thanks for the clarification.
- Rob -

**Comment by gandarez** [comment]
@alanhamlett Is it going to be implemented on wakatime server side, right?

**Comment by gandarez** [comment]
@alanhamlett have you keep in mind it's gonna be at server side? I need to close this PR.

**Comment by alanhamlett** [comment]
@gandarez yes, this will be all server-side so we can close this PR.

**Pull request closed without merging**